src/main/kotlin/gg/essential/elementa/markdown/MarkdownComponent.kt

```diff
@@ -164,8 +182,60 @@ class MarkdownComponent(
         beforeDraw(matrixStack)
 
         val drawState = DrawState(getLeft() - baseX, getTop() - baseY)
+        val parentWindow = Window.of(this)
+
+        drawables.forEach {
 
-        drawables.forEach { it.draw(matrixStack, drawState) }
+            // Draw colored outline around the drawables
+            if (elementaDebug) {
+                if (ScissorEffect.currentScissorState != null) {
+                    GL11.glDisable(GL11.GL_SCISSOR_TEST)
+                }
+
+                val l = it.layout.left.toDouble() + drawState.xShift
+                val r = it.layout.right.toDouble() + drawState.xShift
+                val t = it.layout.top.toDouble() + drawState.yShift
+                val b = it.layout.bottom.toDouble() + drawState.yShift
+
+                val color = getDebugColor(depth(), (parent.hashCode() / PI) % PI)
+
+                // Top outline block
+                UIBlock.drawBlock(
+                    matrixStack,
+                    color,
+                    l - DEBUG_OUTLINE_WIDTH,
+                    t - DEBUG_OUTLINE_WIDTH,
+                    r + DEBUG_OUTLINE_WIDTH,
+                    t
+                )
+
+                // Right outline block
+                UIBlock.drawBlock(matrixStack, color, r, t, r + DEBUG_OUTLINE_WIDTH, b)
+
+                // Bottom outline block
+                UIBlock.drawBlock(
+                    matrixStack,
+                    color,
+                    l - DEBUG_OUTLINE_WIDTH,
+                    b,
+                    r + DEBUG_OUTLINE_WIDTH,
+                    b + DEBUG_OUTLINE_WIDTH
+                )
+
+                // Left outline block
+                UIBlock.drawBlock(matrixStack, color, l - DEBUG_OUTLINE_WIDTH, t, l, b)
+
+                if (ScissorEffect.currentScissorState != null) {
+                    GL11.glEnable(GL11.GL_SCISSOR_TEST)
+                }
+            }
```

Comment:
DRY. Instead of duplicating all this code, move the existing code in UIComponent into its companion object as an internal function taking left, top, right, bottom and parent (you can compute depth from parent) as arguments, and then call that method from here as well.
You can then also re-use the computed l, t, r and b values in below `isAreaVisible` call.